kratos/utilities/dof_utilities/elimination_build_dof_array_utility.h

```diff
@@ -0,0 +1,227 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos
+{
+
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class EliminationBuildDofArrayUtility
+ * @ingroup KratosCore
+ * @brief Utility class for handling the elimination build of DOFs
+ * @details This static class collects all the methods required to
+ * handle the DOFs when doing an elimination build.
+ * @author Ruben Zorrilla
+ */
+class EliminationBuildDofArrayUtility
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of EliminationBuildDofArrayUtility
+    KRATOS_CLASS_POINTER_DEFINITION(EliminationBuildDofArrayUtility);
+
+    /// Definition of the size type
+    using SizeType = std::size_t;
+
+    /// Definition of the index type
+    using IndexType = std::size_t;
+
+    /// DOFs array type definition from ModelPart (note that this is PointerVectorSet<DofType>)
+    using DofsArrayType = ModelPart::DofsArrayType;
+
+    /// DOFs vector type definition from ModelPart (note that this is std::vector<DofType::Pointer>)s
+    using DofsVectorType = ModelPart::DofsVectorType;
+
+    /// Auxilary DOF set type definition
+    using AuxiliaryDofsSetType = std::unordered_set<Node::DofType::Pointer, DofPointerHasher>;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    EliminationBuildDofArrayUtility() = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    static void SetUpDofArray(
+        const ModelPart& rModelPart,
+        DofsArrayType& rDofArray,
+#ifdef USE_LOCKS_IN_ASSEMBLY
+        std::vector<omp_lock_t>& rLockArray,
```

Comment:
Isn't this  super lgacy?, it is not the moment of removing it?, or refactor in the paralel_utilities?

Reply:
This was my first thought, but after a brief discussion with the KTC we decided to keep it. In any case, note that I did separate it to ease an eventual removal. Also note that It populates the entire implementation of the elimination B&S so we'd better remove it at once in the future.